tfjs-backend-webgpu/src/kernels/scatter_webgpu.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {util} from '@tensorflow/tfjs-core';
+import {getCoordsDataType} from '../shader_preprocessor';
+import {getCoordsDataTypeWgsl, getGlobalIndexStringWgsl, getMainHeaderStringWgsl} from '../shader_preprocessor_wgsl';
+import {computeDispatch, flatDispatchLayout} from '../webgpu_util';
+
+import {getUseWgsl, WebGPUProgram} from './webgpu_program';
+
+export class ScatterProgram implements WebGPUProgram {
+  variableNames = ['updates', 'indices', 'defaultValue'];
+  uniforms: string;
+  uniformsWgsl: string;
+  outputShape: number[];
+  userCode: string;
+  shaderKey: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  workGroupSize: [number, number, number] = [64, 1, 1];
+  size: number;
+  useWgsl: boolean;
+  sliceDim: number;
+  shape: number[];
+  indicesSnippet: string;
+  strideString: string;
+  updatesSnippet: string;
+
+  constructor(
+      updateSize: number, sliceDim: number, indicesRank: number,
+      updatesRank: number, strides: number[], shape: number[],
+      summingDupeIndex = true) {
+    this.outputShape = shape;
+    this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    this.dispatch = computeDispatch(
+        this.dispatchLayout, this.outputShape, this.workGroupSize);
+    this.shaderKey = `scatter_${indicesRank}_${updatesRank}_${strides}`;
```

Comment:
`strides` is already a uniform. Not needed to be part of the shaderKey?

Reply:
Done.